integrationtests/tests/fixtures/real.rs

```diff
@@ -3,51 +3,89 @@ use std::io::Cursor;
 use std::ops::Sub;
 use std::path::PathBuf;
 use std::str::FromStr;
-
-use bitcoin::secp256k1;
-use bitcoin::{Address, Transaction};
-use bitcoincore_rpc::Client;
-use bitcoincore_rpc::{Auth, RpcApi};
-use clightningrpc::LightningRPC;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use bitcoin::{secp256k1, Address, Transaction};
+use bitcoincore_rpc::{Auth, Client, RpcApi};
+use cln_rpc::model::requests;
+use cln_rpc::primitives::{Amount as ClnRpcAmount, AmountOrAny};
+use cln_rpc::{ClnRpc, Request, Response};
 use fedimint_api::config::BitcoindRpcCfg;
 use fedimint_api::encoding::Decodable;
 use fedimint_api::Amount;
 use fedimint_wallet::txoproof::TxOutProof;
+use futures::lock::Mutex;
 use lightning_invoice::Invoice;
 
 use crate::fixtures::{BitcoinTest, LightningTest};
 
 pub struct RealLightningTest {
-    rpc_gateway: LightningRPC,
-    rpc_other: LightningRPC,
+    rpc_gateway: Arc<Mutex<ClnRpc>>,
+    rpc_other: Arc<Mutex<ClnRpc>>,
     initial_balance: Amount,
     pub gateway_node_pub_key: secp256k1::PublicKey,
 }
 
+#[async_trait]
 impl LightningTest for RealLightningTest {
-    fn invoice(&self, amount: Amount, expiry_time: Option<u64>) -> Invoice {
+    async fn invoice(&self, amount: Amount, expiry_time: Option<u64>) -> Invoice {
         let random: u64 = rand::random();
-        let invoice = self
+        let invoice_req = requests::InvoiceRequest {
+            msatoshi: AmountOrAny::Amount(ClnRpcAmount::from_msat(amount.milli_sat)),
+            description: "".to_string(),
+            label: random.to_string(),
+            expiry: expiry_time,
+            fallbacks: None,
+            preimage: None,
+            exposeprivatechannels: None,
+            cltv: None,
+            deschashonly: None,
+        };
+
+        let invoice_resp = if let Response::Invoice(data) = self
             .rpc_other
-            .invoice(amount.milli_sat, &random.to_string(), "", expiry_time)
-            .unwrap();
-        Invoice::from_str(&invoice.bolt11).unwrap()
+            .lock()
+            .await
+            .call(Request::Invoice(invoice_req))
+            .await
+            .unwrap()
+        {
+            data
+        } else {
+            panic!("cln-rpc response did not match expected InvoiceResponse")
+        };
+
+        Invoice::from_str(&invoice_resp.bolt11).unwrap()
     }
 
-    fn amount_sent(&self) -> Amount {
+    async fn amount_sent(&self) -> Amount {
         self.initial_balance
-            .sub(Self::channel_balance(&self.rpc_gateway))
+            .sub(Self::channel_balance(self.rpc_gateway.clone()).await)
     }
 }
 
 impl RealLightningTest {
```

Comment:
is there a reason why there are two `impl RealLightningTest` blocks or can they be merged into one?

Reply:
Oh missed this comment, we can do a tiny refactor PR to merge them into one.